lib/mustache.ml

```diff
@@ -142,14 +142,55 @@ let to_string x =
   Buffer.contents b
 
 (* Parsing: produces an ast with locations. *)
+type template_parse_error = {
+  lexbuf: Lexing.lexbuf;
+  kind: error_kind;
+}
+and error_kind = Parsing | Lexing of string
+
+exception Parse_error of template_parse_error
```

Comment:
Should we register a printer for this exception?

Reply:
Good point, I added exception-registration logic in a new commit.